app/models/concrete_slot.rb

```diff
@@ -56,4 +56,8 @@ def duration
   def <=>(other)
     to_s <=> other.to_s
   end
+
+  def overlaps?(other)
+    end_at > other.begin_at
+  end
```

Comment:
No, if s1 is 10:00 to 11:00 and other if 12:00 to 13:00 then `other.end_at > begin_at` is true but it does not overlap.